### PR TITLE
Payload encoding changes

### PR DESCRIPTION
**Pull request by kylecjo**

### CODE REVIEW

**Comment by j-rewerts** [comment]
Could you give me a quick run down of what each of the following packages does for us?
Adafruit-Blinka==0.2.7
adafruit-circuitpython-amg88xx==1.1.0
adafruit-circuitpython-busdevice==2.2.5
adafruit-circuitpython-register==1.3.1
gpsd-py3==0.3.0
pandas==0.23.4
pexpect==4.6.0
ptyprocess==0.6.0
pynmea2==1.12.0
rpi-ws281x==4.0.0
RPi.GPIO==0.6.3

File: requirements.txt [hunk 1]
```diff
@@ -1,22 +1,33 @@
 Adafruit-AMG88xx==1.4.1
```
Comment (j-rewerts) [inline requirements.txt]: Guessing this can be removed now.

File: src/thermal_tracker.py [hunk 2]
```diff
@@ -35,40 +34,58 @@ def map_value(x, in_min, in_max, out_min, out_max):
 
 def transmit(str):
```
Comment (j-rewerts) [inline src/thermal_tracker.py:35]: Do we use this function anymore?

File: src/ttn/thethingsnetwork-send-v1.cpp [hunk 3]
```diff
@@ -136,11 +135,12 @@ static void do_send(osjob_t *j)
         fprintf(stdout, "OP_TXRXPEND, not sending");
     }
     else
-    {
+    {
```
Comment (j-rewerts) [inline src/ttn/thethingsnetwork-send-v1.cpp]: nit

Comment (j-rewerts) [reply src/ttn/thethingsnetwork-send-v1.cpp]: unneeded space

File: src/ttn/thethingsnetwork-send-v1.cpp [hunk 4]
```diff
@@ -136,11 +135,12 @@ static void do_send(osjob_t *j)
         fprintf(stdout, "OP_TXRXPEND, not sending");
     }
     else
-    {
+    {	
         readData();
         unsigned char arr[mydata.size()];
         std::copy(mydata.begin(), mydata.end(), arr);
         LMIC_setTxData2(1, arr, sizeof(arr), 0);
+        std::fill(mydata.begin(),mydata.end(), 0);
```
Comment (j-rewerts) [inline src/ttn/thethingsnetwork-send-v1.cpp]: nit: add space b/w function params.

**Comment by kylecjo** [comment]
pexpect: used for creating the lora process (I don't really know what this does, it was already there when I started)
gpsd-py3: parsing the gps data

adafruit-circuitpython-amg88xx: replacement for the Adafruit-AMG88xx library jeanna used

The following ones are dependencies for adafruit-circuitpython-amg88xx.
Adafruit-Blinka
adafruit-circuitpython-busdevice
adafruit-circuitpython-register
rpi-ws281x
RPi.GPIO

pynmea2: Initially, I was going to use this to parse the raw gps data, but then I found gpsd-py3, so I could get rid of this
pandas: can get rid of this, was experimenting with it
ptyprocess: no idea what this is, not sure if we need it, might be a part of pexpect

I did a `pip freeze > requirements.txt` in the virtualenv to update the requirements.txt but that populates it with everything in the virtualenv - including the modules I experimented with. Do you maybe know a better way to get just the modules we need?

File: src/thermal_tracker.py (hunk 2 above)
Comment (kylecjo) [reply src/thermal_tracker.py:35]: Yea, the send_lora() function wraps this function and calls it every LORA_SEND_INTERVAL seconds. You can see it is called in this snippet

`loraproc = Process(
            target=transmit, name='lora_proc', args=(payload, ))`

File: src/ttn/thethingsnetwork-send-v1.cpp (hunk 3 above)
Comment (kylecjo) [reply src/ttn/thethingsnetwork-send-v1.cpp]: I'll run a formatter on the .cpp file

**Comment by j-rewerts** [comment]
I think you have to remove the packages before doing a pip freeze. In cases where your package had dependencies, I think it actually won't remove those, so you have to manually remove those then.

**Review by j-rewerts — APPROVED** [review]
Looks good!

**Pull request merged**